spec/datadog/core/configuration/options_spec.rb

```diff
@@ -255,6 +255,52 @@
         end
       end
 
+      describe '#using_default?' do
+        subject(:using_default?) { options_object.using_default?(name) }
+
+        let(:name) { :foo }
+
+        context 'when the option is defined' do
+          before { options_class.send(:option, name, meta) }
+
+          let(:meta) { {} }
+
+          context 'and a value is set' do
+            let(:value) { double('value') }
+
+            before { options_object.set_option(name, value) }
+
+            it { is_expected.to be(false) }
+          end
+
+          context 'and a value is not set' do
+            context 'and no value is configured' do
+              it { is_expected.to be(true) }
+            end
+
+            context 'and a default is configured' do
+              let(:meta) { super().merge(default: 'anything') }
+
+              it { is_expected.to be(true) }
+            end
+
+            context 'and an environment variable is configured' do
+              let(:meta) { super().merge(env: 'TEST_ENV_VAR') }
+
+              around do |example|
+                ClimateControl.modify('TEST_ENV_VAR' => 'anything') { example.run }
+              end
+
+              it { is_expected.to be(false) }
+            end
+          end
+        end
+
+        context 'when the option is not defined' do
+          it { expect { using_default? }.to raise_error(described_class::InvalidOptionError) }
+        end
+      end
```

Comment:
```suggestion
describe '#using_default?' do
        subject(:using_default?) { options_object.using_default?(name) }

        let(:name) { :foo }

        context 'when the option is defined' do
          before { options_class.send(:option, name, meta) }

          let(:meta) { {} }

          context 'and a value is set' do
            before { options_object.set_option(name, 'something') }

            it { is_expected.to be(false) }
          end

          context 'and a value is not set' do
            context 'and no default value is configured' do
              it { is_expected.to be(true) }
            end

            context 'and a default value is configured' do
              let(:meta) { { default: 'anything' } }

              it { is_expected.to be(true) }

              context 'and an environment variable is configured' do
                let(:meta) { { default: 'anything', env: 'TEST_ENV_VAR' } }

                context 'and an environmet variable is set' do
                  around do |example|
                    ClimateControl.modify('TEST_ENV_VAR' => 'anything') { example.run }
                  end

                  it { is_expected.to be(false) }
                end

                context 'and an environment variable is not set' do
                  it { is_expected.to be(true) }
                end
              end
            end

            context 'an environment variable is configured' do
              let(:meta) { { env: 'TEST_ENV_VAR' } }

              context 'and an environmet variable is set' do
                around do |example|
                  ClimateControl.modify('TEST_ENV_VAR' => 'anything') { example.run }
                end

                it { is_expected.to be(false) }
              end

              context 'and an environment variable is not set' do
                it { is_expected.to be(true) }
              end
            end
          end
        end
```

The suggestion includes:
- Remove calls to `super` inside `let`
- Changed some test description to be more specific 
- Add a case for when the `env` option is set but the ENV variable is not configured 
- Add a context for `env` is defined without `default` value

Reply:
Also, I wrote some specs on https://github.com/DataDog/dd-trace-rb/blob/37d556d1245e189fa840799f9995384fdb4aaf4a/spec/datadog/core/configuration/base_spec.rb#L145-L218

What do you think about removing those? I feel they test the same thing (without the `env` part) I think your specs are better suited to test the feature fully.

Reply:
Thank you for the updates, applied!

> What do you think about removing those?

I think it's fine to remove those, as they seem a bit redundant like you said. I went ahead and did that.